contracts/pm/mixins/MixinContractRegistry.sol

```diff
@@ -4,7 +4,15 @@ import "../../ManagerProxyTarget.sol";
 import "./interfaces/MContractRegistry.sol";
 
 
-contract MixinContractRegistry is ManagerProxyTarget, MContractRegistry {
+contract MixinContractRegistry is MContractRegistry, ManagerProxyTarget {
```

Comment:
Ah so I'm guessing that the inheritance order is switched because we need to override the `whenSystemNotPaused()` implementation in `MContractRegistry` with the implementation in `ManagerProxyTarget`. Let's add a `@dev` for this contract (i.e. in a NatSpec comment right before this line) that mentions the importance of inheriting `MContractRegistry` *before* `ManagerProxyTarget`.

Reply:
I'm not sure a comment is necessary as the need for the change is part of the solidity documentation on [multiple inheritance and linearization](https://solidity.readthedocs.io/en/v0.5.3/contracts.html#multiple-inheritance-and-linearization)

The order in which the bases are read is right-to-left so if we'd switch this around we'd be calling the modifier in `MContractRegistry` which does nothing but continue the operation.

Reply:
> I'm not sure a comment is necessary as the need for the change is part of the solidity documentation

Fair enough! Though, I'd note that the real reason for not adding a comment is not that multiple inheritance is described in the Solidity docs, but rather that our unit tests ensure that the correct modifier behavior is exhibited when we test the functions that use the modifier so if we ever accidentally switched the inheritance order the tests would fail. While we do have the right test cases setup in this situation, generally we should be cautious around multiple inheritance because the end behavior may not always be so obvious - see this [blog post](https://pdaian.com/blog/solidity-anti-patterns-fun-with-inheritance-dag-abuse/) for a fun deep dive into a sneaky inheritance based backdoor added to a contract.